api/types.proto

```diff
@@ -696,3 +696,9 @@ message ManagerStatus {
 	// Reachability specifies whether this node is reachable.
 	RaftMemberStatus.Reachability reachability = 4;
 }
+
+message Device {
+	string source = 1;
+	string target = 2;
+	string permissions = 3;
```

Comment:
It's actually more complicated than that. If you have 4 GPUs on your machine, the available devices will be:
```
$  ls /dev/nvidia*
/dev/nvidia0  /dev/nvidia1  /dev/nvidia2  /dev/nvidia3  /dev/nvidiactl  /dev/nvidia-modeset  /dev/nvidia-uvm  /dev/nvidia-uvm-tools
```
Devices `/dev/nvidia[0-3]` are the individual GPUs that can be selected, but the others are control devices that **must** be shared between all containers, so it will not fit this class abstraction and will require additional logic.

Other considerations:
- For the sake of completeness, let's list other potential interesting devices apart from GPUs:
disks (`/dev/sd*`), Infiniband (`/dev/infiniband/*`), other types of accelerators (FPGAs, Xeon Phi).
- Not all the GPUs are necessarily the same on a single machine, `/dev/nvidia0` could be 2x faster than `/dev/nvidia1`. While this consideration might be advanced, it's interesting to keep it in mind, if we want to enable that in the future.
- If we consider disks, would it make sense to have a `/dev/sd*` class? Does it make sense to have a service exclusively owning `/dev/sda`? I'm not sure the "class" abstraction could extend well to other types of devices.

The question of supporting additional resources and having a scheduler aware of these resources is a complex one. So, here's what I'm suggesting:
1) We should allow devices in services, because a services might need a device, any device, no questions asked. It will be the responsibility of the user to make sure the device exists on all nodes.
2) We can start a longer discussion regarding supporting additional resource types (not only GPUs) in Swarm.